promql/engine_test.go

```diff
@@ -634,8 +648,10 @@ load 10s
 			Start: time.Unix(10, 0),
 		},
 		{
-			Query:      "metric[20s:5s]",
-			MaxSamples: 3,
+			Query:        "metric[20s:5s]",
+			MaxSamples:   3,
+			TotalSamples: 3,
```

Comment:
This feels low for a subquery, what does a more complex subquery produce?

To test this properly we should also add in some aggregators, binary operators, and functions on instant vectors.

Reply:
Ran instant query (after a min of prometheus start) for the following:

`go_goroutines[5m:10s]` => 13 (verified via console)

`go_goroutines[5m:10s]` => 201 (verified in console)

Reply:
@brian-brazil does it look satisfactory now?

Reply:
There still needs to be a complex subquery.